targets/photos/services/onPhotoUpload.js

```diff
@@ -1,5 +1,76 @@
+import { log, cozyClient } from 'cozy-konnector-libs'
+
+import { DOCTYPE_FILES } from '../../../src/drive/lib/doctypes'
+import {
+  readSetting,
+  defaultParameters,
+  createDefaultSetting
+} from '../../../src/photos/ducks/clustering/settings'
+
+// Returns the photos metadata sorted by date
+const extractInfo = photos => {
+  let info = photos
+    .map(file => {
+      const photo = {
+        id: file._id,
+        name: file.name
+      }
+      if (file.metadata) {
+        photo.date = file.metadata.datetime
+        photo.gps = file.metadata.gps
+      } else {
+        photo.date = file.created_at
+      }
+      const hours = new Date(photo.date.slice(0, 19)).getTime() / 1000 / 3600
```

Comment:
So we want a timestamp expressed in hours?

Also, why do we have to extract a part of the `photo.date` string? What else is there in it?

Reply:
Yes we use the hours as the unit for our temporal metric, as it easier to understand use than milliseconds in our case. 
For the `photo.date`, it is useful after the clustering, we need to save the period of a newly created album (dates of first and last photos), so we can use the ones already saved in the file.

Reply:
Ok, thanks for confirming the hour-timestamp, it's not usual but it makes sense!

For the slice, my main concern is that `0, 19` looks very arbitrary and without looking at one of these dates, it's hard to figure out what is happening. What happens if we don't slice the string at all? Can we still use it in `new Date()`?

Reply:
The date format generated by the stack is like `2018-11-15T00:07:19.519852762+01:00`. So the `0,19` slice was to keep the precision to the second. However, I admit this is not very useful, and more importantly, I realize the `+01:00` might be important in case of different timezone. So I'll remove the slice :)